lib/plugins/aws/package/lib/packageService.js

```diff
@@ -54,6 +57,8 @@ module.exports = {
     const include = this.getIncludes();
     const zipFileName = this.provider.naming.getServiceArtifactName();
 
+    _.set(this.serverless.service, 'package.artifact', zipFileName);
```

Comment:
Ok. this is one part of the solution... The second one would be to replace the persisted path part of the artifact with the actual path for the deploy phase in `extendedValidate()`.

So extendedValidate() should do a `service.package.artifact = path.join(this.serverless.config.servicePath, '.serverless', << filename part of state.package.artifact >>)`